src/components/campaignguide/ScenarioStepComponent/NarrationStepComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useCallback, useContext, useState } from "react";
+import { View } from "react-native";
+import { Icon, Text } from "react-native-elements";
+import { useSelector } from "react-redux";
+
+import { StyleContext } from "@styles/StyleContext";
+import TrackPlayer, {
+  STATE_PLAYING,
+  useTrackPlayerEvents,
+} from "react-native-track-player";
+
+import space, { s } from "@styles/space";
+import { hasDissonantVoices } from "@reducers";
+import { playNarration } from "@components/campaignguide/Narrator";
+import { Narration } from "@data/scenario/types";
+import { SHOW_DISSONANT_VOICES } from "@app_constants";
+
+interface CheckProps {
+  narration?: Narration;
+  children: JSX.Element;
+}
+
+export function CheckDissonantVoicesComponent(props: CheckProps) {
+  const { children, narration } = props;
+  if (!SHOW_DISSONANT_VOICES || narration === undefined) return <></>;
+
+  const hasDS = useSelector(hasDissonantVoices);
+  return (hasDS ? children : <></>);
+}
+
+interface IconProps {
+  narration: Narration;
+}
+
+export function NarrationStatusButton(props: IconProps) {
+  const { narration } = props;
+
+  const [playerState, setPlayerState] = useState<string | number | null>(null);
+  const [currentTrackState, setCurrentTrackState] = useState<string | null>(null);
+  
+  TrackPlayer.getState().then(state => setPlayerState(state));
```

Comment:
Should this be in a `useEffect` of some kind?  As currently written I believe it will execute on every render?  If it does need to be run on every render, I think putting it in a `useEffect` is still preferrable to the state updates like this.

Reply:
If the intent is to only run it once on mount, correct construct is

```
useEffect(() => {
  TrackPlayer.getState().then(state => setPlayerState(state));
  TrackPlayer.getCurrentTrack().then(currentTrack => setCurrentTrackState(currentTrack));
} ,[]);
```

With a eslint ignore on the dependencies.

Reply:
I will look into how best to use a promise using hooks. It should only run to initiate `playerState` and `currentTrackState`. After that `useTrackPlayerEvents` should handle the state changes.